File: spec/java/src/io/kaitai/struct/spec/TestSwitchManualIntSize.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.SwitchManualIntSize;
import org.testng.annotations.Test;

import static org.testng.Assert.assertEquals;

public class TestSwitchManualIntSize extends CommonSpec {
    @Test
    public void testSwitchManualIntSize() throws Exception {
        SwitchManualIntSize r = SwitchManualIntSize.fromFile(SRC_DIR + "switch_tlv.bin");

        assertEquals(r.chunks().size(), 4);

        assertEquals(r.chunks().get(0).code(), 0x11);
        SwitchManualIntSize.Chunk.ChunkMeta meta = (SwitchManualIntSize.Chunk.ChunkMeta) r.chunks().get(0).body();
        assertEquals(meta.title(), "Stuff");
        assertEquals(meta.author(), "Me");

        assertEquals(r.chunks().get(1).code(), 0x22);
        assertEquals(
                ((SwitchManualIntSize.Chunk.ChunkDir) r.chunks().get(1).body()).entries().toArray(),
                new String[] { "AAAA", "BBBB", "CCCC" }
        );

        assertEquals(r.chunks().get(2).code(), 0x33);
        assertEquals(r.chunks().get(2).body(), new byte[] { 0x10, 0x20, 0x30, 0x40, 0x50, 0x60, 0x70, (byte) 0x80 });

        assertEquals(r.chunks().get(3).code(), 0xff);
        assertEquals(r.chunks().get(3).body(), new byte[] {});
    }
}

File: spec/javascript/test_switch_manual_int_size.js
var assert = require('assert');
var testHelper = require('testHelper');

testHelper('SwitchManualIntSize', 'src/switch_tlv.bin', function(r) {
  assert.equal(r.chunks.length, 4);

  assert.equal(r.chunks[0].code, 0x11);
  meta = r.chunks[0].body
  assert.equal(meta.title, 'Stuff');
  assert.equal(meta.author, 'Me');

  assert.equal(r.chunks[1].code, 0x22);
  assert.deepEqual(r.chunks[1].body.entries, ['AAAA', 'BBBB', 'CCCC']);

  assert.equal(r.chunks[2].code, 0x33);
  assert.equal(r.chunks[2].body.toString(), [0x10, 0x20, 0x30, 0x40, 0x50, 0x60, 0x70, 0x80].toString());

  assert.equal(r.chunks[3].code, 0xff);
  assert.equal(r.chunks[3].body, '');
});
